CBForest/KeyStore.cc

```diff
@@ -143,10 +143,10 @@ namespace forestdb {
             memcpy(keybuf, key.buf, key.size);
             key.buf = keybuf;
         }
-        fdb_doc doc = {
-            .key = (void*)key.buf,
-            .keylen = key.size,
-        };
+        fdb_doc doc;
```

Comment:
Again, this needs to be `fdb_doc doc = {}` — this time it's crucial, as your current change leaves most of the fields as garbage!